src/coreComponents/constitutive/fluid/CompressibleSinglePhaseFluid.cpp

```diff
@@ -114,35 +168,48 @@ void CompressibleSinglePhaseFluid::postProcessInput()
   checkPositive( m_referenceDensity, viewKeyStruct::referenceDensityString() );
   checkPositive( m_referenceViscosity, viewKeyStruct::referenceViscosityString() );
 
-  // Due to the way update wrapper is currently implemented, we can only support one model type
-  auto const checkModelType = [&]( ExponentApproximationType const value, auto const & attribute )
+  if ( m_isThermal )
   {
-    GEOSX_THROW_IF_NE_MSG( value, ExponentApproximationType::Linear,
-                           GEOSX_FMT( "{}: invalid model type in attribute '{}' (only linear currently supported)", getFullName(), attribute ),
-                           InputError );
-  };
-  checkModelType( m_densityModelType, viewKeyStruct::densityModelTypeString() );
-  checkModelType( m_viscosityModelType, viewKeyStruct::viscosityModelTypeString() );
-
-  // Set default values for derivatives (cannot be done in base class)
-  // TODO: reconsider the necessity of this
-
-  real64 dRho_dP;
-  real64 dVisc_dP;
-  createKernelWrapper().compute( m_referencePressure, m_referenceDensity, dRho_dP, m_referenceViscosity, dVisc_dP );
-  getExtrinsicData< extrinsicMeshData::singlefluid::dDensity_dPressure >().setDefaultValue( dRho_dP );
-  getExtrinsicData< extrinsicMeshData::singlefluid::dViscosity_dPressure >().setDefaultValue( dVisc_dP );
+    checkNonnegative( m_referenceInternalEnergy, viewKeyStruct::referenceInternalEnergyString() ); 
+  }
+
+  // // Due to the way update wrapper is currently implemented, we can only support one model type
+  // auto const checkModelType = [&]( ExponentApproximationType const value, auto const & attribute )
+  // {
+  //   GEOSX_THROW_IF_NE_MSG( value, ExponentApproximationType::Linear,
+  //                          GEOSX_FMT( "{}: invalid model type in attribute '{}' (only linear currently supported)", getFullName(), attribute ),
+  //                          InputError );
+  // };
+  // checkModelType( m_densityModelType, viewKeyStruct::densityModelTypeString() );
+  // checkModelType( m_viscosityModelType, viewKeyStruct::viscosityModelTypeString() );
+
+  // // Set default values for derivatives (cannot be done in base class)
+  // // TODO: reconsider the necessity of this
+
+  // real64 dRho_dP;
+  // real64 dVisc_dP;
+  // createKernelWrapper().compute( m_referencePressure, m_referenceDensity, dRho_dP, m_referenceViscosity, dVisc_dP );
+  // getExtrinsicData< extrinsicMeshData::singlefluid::dDensity_dPressure >().setDefaultValue( dRho_dP );
+  // getExtrinsicData< extrinsicMeshData::singlefluid::dViscosity_dPressure >().setDefaultValue( dVisc_dP );
```

Comment:
Why this commented code? Maybe it's still WIP?

Reply:
> Why this commented code? Maybe it's still WIP?

Yes. I temporarily commented the model type checks and default values setup for the derivatives. For the model type check, I will modify the code later as the original version only permit the linear type. For setting default valus of derivatives, I saw the comment in "TODO", so I am not quite sure whether it is necessary. So for now, I just commented them.